game/shared/tf/tf_playeranimstate.cpp

```diff
@@ -1344,13 +1344,19 @@ bool CTFPlayerAnimState::HandleJumping( Activity &idealActivity )
 		
 	// Handle air walking before handling jumping - air walking supersedes jump
 	TFPlayerClassData_t *pData = m_pTFPlayer->GetPlayerClass()->GetData();
-	bool bValidAirWalkClass = ( pData && pData->m_bDontDoAirwalk == false );
+	const bool bValidAirWalkClass = ( pData && pData->m_bDontDoAirwalk == false );
 
-	if ( bValidAirWalkClass && ( vecVelocity.z > 300.0f || m_bInAirWalk || m_pTFPlayer->GetGrapplingHookTarget() != NULL ) && !bInDuck )
+	if ( bValidAirWalkClass && ( vecVelocity.z > 300.0f || m_bInAirWalk || m_pTFPlayer->GetGrapplingHookTarget() != NULL ) )
 	{
-		// Check to see if we were in an airwalk and now we are basically on the ground.
-		if ( ( GetBasePlayer()->GetFlags() & FL_ONGROUND ) && m_bInAirWalk )
-		{				
+		if ( ( GetBasePlayer()->GetFlags() & FL_ONGROUND ) == 0 )
+		{
+			// In an air walk.
+			idealActivity = ACT_MP_AIRWALK;
```

Comment:
Only do an airwalk on `!bInDuck`?

Reply:
This probably makes it so a crouched rocket-jumping player is doing the airwalk animation when they should not be.

Reply:
Shouldn't a crouched rocket jumping playing be airwalking? As far as I can tell there is no crouched airwalk animation in the models, which would be ideal to use for this case. I feel like it would look better if all players play the airwalk animation if they're rocket jumping instead of remaining crouched. 

Would like some input on this.

Reply:
Does air walk only mess with the feet?